### ISSUE TITLE
Overview of unused and undocumented attributes

### ISSUE THREAD

**Issue by Scott-Guest** [runtimeverification/k#3500]
A number of attributes are unused, either because they 

- are undocumented and never appear in any K source files in any downstream semantics
- do appear in K source files or are added programmatically, but are never consumed anywhere

We should remove all these unused attributes. 

From a simple text search, an initial list is as follows:

| Attribute | Read | Written | Documented |
|-----------|------|---------|------------|
| alias | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| alias-rec | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| all-path | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| anywhere | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| applyPriority | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| assoc | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| avoid | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| bag | :heavy_check_mark: | :heavy_check_mark: | :x: |
| binder | :heavy_check_mark: | :heavy_check_mark: | :medical_symbol: |
| bitwidth | :x: | :x: | :x: |
| bracket | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| bracketLabel | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| cell | :heavy_check_mark: | :heavy_check_mark: | :x: |
| cellFragment | :heavy_check_mark: | :heavy_check_mark: | :x: |
| cellName | :heavy_check_mark: | :heavy_check_mark: | :x: |
| cellOptAbsent | :heavy_check_mark: | :heavy_check_mark: | :x: |
| color | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| colors | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| comm | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| concrete | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| context | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| cool | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| digest | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| element | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| exponent | :x: | :x: | :x: |
| format | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| freshGenerator | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| function | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| functional | :heavy_check_mark: | :heavy_check_mark: | :x: |
| generatedByListSubsorting | :heavy_check_mark: | :x: | :x: |
| heat | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| hybrid | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| hook | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| idem | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| impure | :heavy_check_mark: | :heavy_check_mark: | :x: |
| initializer | :heavy_check_mark: | :heavy_check_mark: | :x: |
| injective | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| internal | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| kast | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| klabel | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| kore | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| label | :heavy_check_mark: | :heavy_check_mark: | :x: |
| latex | :heavy_check_mark: | :heavy_check_mark: | :medical_symbol: |
| left | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| lemma | :heavy_check_mark: | :x: | :x: |
| org.kframework.attributes.Location | :heavy_check_mark: | :heavy_check_mark: | :x: |
| locations | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| macro | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| macro-rec | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| maincell | :heavy_check_mark: | :heavy_check_mark: | :x: |
| matching | :x: | :x: | :x: |
| memo | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| non-assoc | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| non-executable | :heavy_check_mark: | :heavy_check_mark: | :x: |
| noThread | :x: | :heavy_check_mark: | :x: |
| one-path | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| originalPrd | :heavy_check_mark: | :heavy_check_mark: | :x: |
| owise | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| pattern | :heavy_check_mark: | :x: | :x: |
| pattern-folding | :heavy_check_mark: | :x: | :x: |
| prec | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| predicate | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| prefer | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| priority | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| private | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| org.kframework.definition.Production | :heavy_check_mark: | :heavy_check_mark: | :x: |
| proj | :x: | :x: | :x: |
| public | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| recordPrd | :heavy_check_mark: | :heavy_check_mark: | :x: |
| refers_RESTORE_CONFIGURATION | :x: | :x: | :x: |
| refers_THIS_CONFIGURATION | :x: | :x: | :x: |
| result | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| returnsUnit | :x: | :heavy_check_mark: | :x: |
| right | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| seqstrict | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| significand | :x: | :x: | :x: |
| simplification | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| smtlib | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| smt-hook | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| smt-lemma | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| smt-prelude | :heavy_check_mark: | :heavy_check_mark: | :x: |
| org.kframework.attributes.Source | :heavy_check_mark: | :heavy_check_mark: | :x: |
| specification | :x: | :x: | :x: |
| stream | :heavy_check_mark: | :heavy_check_mark: | :medical_symbol: |
| strict | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| symbol | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| symbolic | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| syntaxModule | :heavy_check_mark: | :heavy_check_mark: | :x: |
| tag | :heavy_check_mark: | :heavy_check_mark: | :x: |
| token | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| topcell | :x: | :heavy_check_mark: | :x: |
| topRule | :x: | :x: | :x: |
| total | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| transition | :x: | :heavy_check_mark: | :x: |
| trusted | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| unblock | :heavy_check_mark: | :heavy_check_mark: | :x: |
| unboundVariables | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| unit | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| UNIQUE_ID | :heavy_check_mark: | :heavy_check_mark: | :x: |
| unparseAvoid | :heavy_check_mark: | :heavy_check_mark: | :x: |
| unused | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| userList | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |

The :medical_symbol: mark indicates that the attribute was mentioned in the documentation but not fully explained. Some of these are internal-only and may not need documentation.

**Comment by Baltoli** [comment]
Closing as superseded by #3946; can reopen if there's a gap between the information in the two PRs.
